core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -975,12 +978,13 @@ class ReplicaManager(val config: KafkaConfig,
 
   /**
    *
-   * @param topicPartition  the topic partition to maybe verify
-   * @param transactionalId the transactional id for the transaction
-   * @param producerId      the producer id for the producer writing to the transaction
-   * @param producerEpoch   the epoch of the producer writing to the transaction
-   * @param baseSequence    the base sequence of the first record in the batch we are trying to append
-   * @param callback        the method to execute once the verification is either completed or returns an error
+   * @param topicPartition      the topic partition to maybe verify
+   * @param transactionalId     the transactional id for the transaction
+   * @param producerId          the producer id for the producer writing to the transaction
+   * @param producerEpoch       the epoch of the producer writing to the transaction
+   * @param baseSequence        the base sequence of the first record in the batch we are trying to append
+   * @param callback            the method to execute once the verification is either completed or returns an error
+   * @param partitionOperation  the enum value which determines whether the client is ready to handle the new Abortable Txn Errors
```

Comment:
nit: to make this more generic we should say something like "determines which version of the client sent the request so the addPartitionsToTxnManager handles it appropriately".

I also wonder if we want to rename the argument. PartitionOperation is a little unclear.